packages/ts/react-signals/test/Signals.spec.tsx

```diff
@@ -59,5 +60,27 @@ describe('@vaadin/hilla-react-signals', () => {
       await nextFrame();
       expect(result.container.textContent).to.equal('Value is 42');
     });
+
+    it('should set the underlying value locally without waiting for server confirmation', () => {
+      const numberSignal = new NumberSignal(publishSpy);
+      expect(numberSignal.value).to.equal(undefined);
+      numberSignal.value = 42;
+      expect(numberSignal.value).to.equal(42);
+
+      const anotherNumberSignal = new NumberSignal(publishSpy);
+      let counter = 0;
+      effect(() => {
+        if (counter === 0) {
+          expect(anotherNumberSignal.value).to.be.undefined;
+        } else if (counter === 1) {
+          expect(anotherNumberSignal.value).to.equal(42);
+        } else {
+          expect(anotherNumberSignal.value).to.equal(43);
+        }
+        counter += 1;
+      });
```

Comment:
I'm not sure if this code is evaluated because the test is probably over before the effect starts working. You have to add an expectation in the end of the test and outside of any callback. You could use an array or a map to collect data, and in the end of the test you can assert it.

Reply:
I know that the effect is called since if you change the expected values it will fail, but I see your point ;) 
I'll simply expect the counter's value to be 2 by the end of test, that should be enough I guess.